ProtocolHandler.pm

```diff
@@ -36,7 +52,165 @@ sub new {
+# no data could be returned before the first chunk is here.
+# If this method is omitted, then player.pm will try to calculate the value using 
+# bitrate and buffersec preference (that is also used to size the buffer) resulting
+# in a too huge value, topped at 255, that is good for mp3 but maybe little for flac.
+#
+# THE FOLLOWING SEEMS NOT TO WORK...
+#
+
+sub bufferThreshold{
+	my ($class, $client, $url) = @_;
+
+    #We have bufferThreshold in prefs, let's use it.
+    my $clientPrefs= $prefs->client($client);
+    my $bufferThreshold= $clientPrefs->get('bufferThreshold');
+    
+    my $bufferSecs = $prefs->get('bufferSecs') || BUFFERING_SECONDS;
+    #limit to BUFFERING_SECONDS seconds.
+    if ($bufferSecs > BUFFERING_SECONDS){$bufferSecs = BUFFERING_SECONDS;}
+    
+    my $format = $class->getFormatForURL($url);
+    
+	return ($format eq 'flc' ? 80 : 32) * $bufferSecs;
+}
+
+sub sysread {
+	my $self = $_[0];
+    #my $chunk = $_[1];
+	my $chunkSize = $_[2];
+
+	my $metaInterval = ${*$self}{'metaInterval'};
+	my $metaPointer  = ${*$self}{'metaPointer'};
+    
+    #Data::Dump::dump("Plugins::Qobuz::ProtocolHandler - sysread: ", $metaInterval, $metaPointer, $chunkSize);
+    my $readLength;
+    
+	if ($metaInterval && ($metaPointer + $chunkSize) > $metaInterval) {
+
+		$chunkSize = $metaInterval - $metaPointer;
+
+		# This is very verbose...
+		#$log->debug("Reduced chunksize to $chunkSize for metadata");
+        Data::Dump::dump("Reduced chunksize for metadata", $chunkSize);
+        
+        $readLength = CORE::sysread($self, $_[1], $chunkSize, length($_[1] || '' ))
+
+    } else {
+        
+        $readLength = _sysread($self, $_[1], MIN_OUT);
+    }
+    
+    #Data::Dump::dump("sysread: ", $_[1], $chunkSize, length($_[1]));
+	#my $readLength = CORE::sysread($self, $_[1], $chunkSize, length($_[1] || '' ));
+
+	if ($metaInterval && $readLength) {
+
+		$metaPointer += $readLength;
+		${*$self}{'metaPointer'} = $metaPointer;
+
+		# handle instream metadata for shoutcast/icecast
+		if ($metaPointer == $metaInterval) {
+
+			$self->readMetaData();
+
+			${*$self}{'metaPointer'} = 0;
+
+		} elsif ($metaPointer > $metaInterval) {
+
+			main::DEBUGLOG && $log->debug("The shoutcast metadata overshot the interval.");
+		}	
+	}
+
+	return $readLength;
+}
+
+sub _sysread(){
+    use bytes;
+    
+    my $self = $_[0];
+    
+    my $v = $self->vars;
+    my $url = ${*$self}{'url'};
+
+    # need more data
+	if ( length $v->{'outBuf'} < MAX_OUT && !$v->{'fetching'}) {
+        
+        my $range;
+        
+        if ($v->{'streaming'}){
+            $range = "bytes=$v->{offset}-" . ($v->{offset} + RANGE_SIZE - 1);
+ 
+        } else {
+            
+            $range = "bytes=$v->{offset}-" . ($v->{offset} + 1);
+        }
+        $v->{'fetching'} = 1;
+        Data::Dump::dump("* Going to fetch:  ", $url, $range, length($v->{'inBuf'} || ''), $v->{'fetching'},$v->{'streaming'});
+						
+		Slim::Networking::SimpleAsyncHTTP->new(
```

Comment:
Whitespace... please fix the formatting. It's so f... up I don't even want to try to understand what belongs together...